app/pages/lab/actions/workflow.js

```diff
@@ -1,14 +1,28 @@
 var apiClient = require('panoptes-client/lib/api-client');
 
+var projects = apiClient.type('projects');
 var workflows = apiClient.type('workflows');
 
 var workflowActions = {
-  createWorkflowForProject: function(projectID, workflowData) {
+  createWorkflowForProject: function(project, workflowData) {
     var allWorkflowData = Object.assign({
-      links: {project: projectID}
+      links: {project: project.id}
     }, workflowData);
 
     return workflows.create(allWorkflowData).save();
+  },
+  copyWorkflowForProject: async function(project, workflowData) {
+    await project.addLink('workflows', [workflowData.id]);
+    project.uncache()
+    const newProject = await projects.get(project.id);
+    const oldLinks = new Set(project.links.workflows);
```

Comment:
you'll need to get the old links before running the update as i believe the json api client will take the resource response from the `project.addLink('workflows', [workflowData.id]);` and update it's cached copy with the new copied workflow id in it.  

So if you do it in this order the two links arrays will match and we won't get the newly copied workflow id out of the response.

Reply:
Good point. `addLink` makes a GET request to refresh the project too. Hidden side effects like this make the old API client a bit difficult to work with.

https://github.com/zooniverse/json-api-client/blob/f4cdb5498df385b38b3bd51a11ed80d44f14ea31/src/resource.coffee#L183-L191

Reply:
Should be fixed now. I've updated the variable naming to use `oldLinks` and `newLinks` too. The new workflow ID is showing up in the API response too.